src/main/java/org/testcontainers/vault/VaultContainer.java

```diff
@@ -0,0 +1,139 @@
+package org.testcontainers.vault;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.LinkableContainer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.testcontainers.shaded.com.github.dockerjava.api.model.Capability.IPC_LOCK;
+
+/**
+ * GenericContainer subclass for Vault specific configuration and features. The main feature is the
+ * withSecretInVault method, where users can specify which secrets to be pre-loaded into Vault for
+ * their specific test scenario.
+ *
+ * Other helpful features include the withVaultPort, and withVaultToken methods for convenience.
+ */
+public class VaultContainer<SELF extends VaultContainer<SELF>> extends GenericContainer<SELF>
+        implements LinkableContainer {
+
+    private boolean preloadSecret = false;
```

Comment:
do we really need this flag? Can't we just assume that if secretsMap is not empty then we need to preload secrets? :)